### PR TITLE
feat: add strategies deployment script

### PR DESCRIPTION
**Pull request by 0xOneTony**
Closes DTT-296

### CODE REVIEW

**Review by ilpepepig — APPROVED** [review]

File: script/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+4. Run the deployment script with:
+```
+script/strategies/deployStrategy.sh ${CHAIN_NAME} ${STRATEGY_NAME}
+```
+For example:
+```
+script/strategies/deployStrategy.sh sepolia DeployDirectAllocation
+```
+
+# Supported networks
```
Comment (0xOneTony) [inline script/README.md:28]: Can we also add a section with `Supported strategy names`

File: script/strategies/deployStrategy.sh [hunk 2]
```diff
@@ -0,0 +1,138 @@
+#!/bin/bash
+
+# Function to dynamically map chains to variables
+# Relevant issue that would deprecate this script if solved: https://github.com/foundry-rs/foundry/issues/7726
+# TODO: add networks
+deploy() {
+  chain=$1
+  strategy=$2
+  case "$chain" in
```
Comment (0xOneTony) [inline script/strategies/deployStrategy.sh:9]: Missing local enviroment

File: script/README.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+# Usage
+
+1. Install dependencies by running `bun install`.
+2. Compile contracts by running `bun smock && bun compile`.
+3. Fill the necessary environment variables in `.env` file. You can use `.env.example` as a template.
+4. Run the deployment script with:
+```
+script/strategies/deployStrategy.sh ${CHAIN_NAME} ${STRATEGY_NAME}
+```
+For example:
```
Comment (0xOneTony) [inline script/README.md:14]: Would be good to also have a detailed example on deploying locally

File: script/README.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+```
+script/strategies/deployStrategy.sh sepolia DirectAllocation
+```
+Or deploying to `local` environment (`127.0.0.1:8545`):
```
Comment (0xOneTony) [inline script/README.md]: Lets also mention that they need to spin up anvil

**Review by 0xOneTony — COMMENTED** [review]
LGTM!

**Review by ilpepepig — APPROVED** [review]

**Review by thelostone-mc — APPROVED** [review]

**Pull request merged**